Automattic/jetpack
# VideoPress + shortcode: Add support for preload/preloadcontent attribute

This PR adds support for the `preloadContent` player attribute to the `videopress` and `wpvideo` shortcode. I'm not sure how this has been missing for so long considering how long it has been supported in the player. I vaguely recall it existing at some point, but maybe some old VideoPress code was removed.

Fixes # https://github.com/Automattic/greenhouse/issues/1554

## Proposed changes:

* Adds support for the `preload` (or `preloadcontent`) video attribute to the `videopress` (and `wpvideo` when videopress is enabled)

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?


Nope

## Testing instructions:






* Test on a jetpack test site with VideoPress enabled
* add a VideoPress shortcode to your site, include the `preload` attribute (ex `[wpvideo gSmUdNwi preload=none]`)
* View the public page, inspect the video. You should see the `preload=none` attribute on the `<video>` tag inside the iframe
* change the `preload` attribute to `preloadcontent`, publish the change
* view the public page, inspect the video. You should still see the `preload=none` attribute on the video
* Do the above test plan separately twice, with either the `Jetpack` plugin activated or the `Jetpack VideoPress` plugin activated
